include/nanobind/eigen/dense.h

```diff
@@ -241,6 +249,107 @@ struct type_caster<Eigen::Ref<T, Options, StrideType>> {
     operator Ref() { return Ref(caster.operator Map()); }
 };
 
+/// Caster for Eigen::SparseMatrix<T>
+template <typename T> struct type_caster<T, enable_if_t<is_eigen_sparse_v<T>>> {
+    using Scalar = typename T::Scalar;
+    using StorageIndex = std::remove_reference_t<decltype(*std::declval<T>().outerIndexPtr())>;
+    using Index = typename T::Index;
+    static constexpr bool rowMajor = T::IsRowMajor;
+
+    using ScalarCaster = make_caster<tensor<Scalar>>;
+    using StorageIndexCaster = make_caster<tensor<StorageIndex>>;
+    using IndexCaster = make_caster<Index>;
+
+    NB_TYPE_CASTER(T, const_name<rowMajor>("scipy.sparse.csr_matrix[",
+					   "scipy.sparse.csc_matrix[")
+		   + make_caster<Scalar>::Name + const_name("]"));
+
+    ScalarCaster scalar_caster;
+    StorageIndexCaster storageindex_caster;
+    IndexCaster index_caster;
+
+    bool from_python(handle src, uint8_t flags, cleanup_list *cleanup) noexcept {
+	if (!src) {
+	    return false;
+	}
+
+	auto obj = borrow<object>(src);
+	object sparse_module = module_::import_("scipy.sparse");
+	object matrix_type = sparse_module.attr(rowMajor ? "csr_matrix" : "csc_matrix");
+
+	if (!obj.type().is(matrix_type)) {
+            try {
+                obj = matrix_type(obj);
+            } catch (const python_error &) {
+                return false;
+            }
+        }
+
+	if (!scalar_caster.from_python(obj.attr("data"), flags, cleanup)) {
+	    return false;
+	}
+        auto values = scalar_caster.value;
+
+	if (!storageindex_caster.from_python(obj.attr("indices"), flags, cleanup)) {
+	    return false;
+	}
+        auto innerIndices = storageindex_caster.value;
+
+	if (!scalar_caster.from_python(obj.attr("indptr"), flags, cleanup)) {
+	    return false;
+	}
+        auto outerIndices = storageindex_caster.value;
+
+	// FIXME: Write with caster.from_python
+	auto shape = cast<std::tuple<Index, Index>>(obj.attr("shape"));
```

Comment:
I am surprised that this works, because the `std::tuple` type caster hasn't been imported (and that seems like a rather heavy dependency). We really don't want to pull in any STL type casters since that may lead to unexpected consequences in user code.

Can you do it manually, e.g.:
```cpp
object shape_o = obj.attr("shape"), nnz_o = obj.attr("nnz");
int rows, cols, nnz;
try {
    if (len(shape_o) != 2)
       return false;
    rows = cast<int>(shape_o[0]);
    cols = cast<int>(shape_o[1]);
    nnz = cast<Index>(nnz_o);
} catch (const python_error &) {
    return false;
}
```